dockerfiles/base/tests/test_base.sh

```diff
@@ -19,15 +19,39 @@ fi
 mkdir $TESTRUN_DIR -p
 
 kill_running_named_container() {
-  if [ $(docker ps --format '{{.Names}}' | grep $1 | wc -l) -eq 1 ]; then
+  if [[ $(docker ps --format '{{.Names}}' | grep $1 | wc -l) -eq 1 ]]; then
     echo "Stopping named container $1"
     docker kill $1 1>/dev/null
   fi
 }
 
 remove_named_container() {
-  if [ $(docker ps -a --format '{{.Names}}' | grep $1 | wc -l) -eq 1 ]; then
+  if [[ $(docker ps -a --format '{{.Names}}' | grep $1 | wc -l) -eq 1 ]]; then
     echo "Removing named container $1"
     docker rm $1 1>/dev/null
   fi
 }
+
+# Pass a port as an argument to check whether is free or not
```

Comment:
I don't think so. The test_base.sh is parent of all tests, if I would move all utility functions to a library_tests.sh, than the test_base.sh would have only few constants in header, sourced library and nothing else (no functions). It would be a transitive sourcing and not needed, I think.